### PR TITLE
FALCON-2030 : Enfornce time partition in the storage path for feeds w…

### PR DESCRIPTION
**Pull request by vramachan**
Enforce time partition in the storage path for feeds with import and export policy.
HCAT storage types, this restriction is already handled.

### CODE REVIEW

File: common/src/test/java/org/apache/falcon/entity/parser/FeedEntityParserTest.java [hunk 1]
```diff
@@ -1160,6 +1160,24 @@ public void testImportFeedSqoopInvalid() throws Exception {
         Assert.fail("ValidationException should have been thrown");
     }
 
+    @Test (expectedExceptions = {ValidationException.class, FalconException.class})
+    public void testImportFeedWithNoTimePartition() throws Exception {
+
+        InputStream feedStream = this.getClass()
+                .getResourceAsStream("/config/feed/feed-import-no-timepartition-0.1.xml");
+        parser.parseAndValidate(feedStream);
+        Assert.fail("ValidationException should have been thrown");
```
Comment (ajayyadava) [inline common/src/test/java/org/apache/falcon/entity/parser/FeedEntityParserTest.java]: This should already be handled by expectedExceptions, right?

File: common/src/test/java/org/apache/falcon/entity/parser/FeedEntityParserTest.java [hunk 2]
```diff
@@ -1160,6 +1160,24 @@ public void testImportFeedSqoopInvalid() throws Exception {
...
+    }
+
+    @Test (expectedExceptions = {ValidationException.class, FalconException.class})
```
Comment (ajayyadava) [inline common/src/test/java/org/apache/falcon/entity/parser/FeedEntityParserTest.java]: Shouldn't this be just ValidationException.class. Using two exceptions make the test a bit indeterministic and may pass in other cases as well.

**Comment by ajayyadava** [comment]
It will be useful to mention it in the docs( entity specification of feed) that the pattern is compulsory for the following scenarios along with some examples of valid and invalid patterns.

File: common/src/test/java/org/apache/falcon/entity/parser/FeedEntityParserTest.java (hunk 1 above)
Comment (vramachan) [inline common/src/test/java/org/apache/falcon/entity/parser/FeedEntityParserTest.java]: @ajayyadava fixed expected exceptions and added documentation.

**Comment by yzheng-hortonworks** [comment]
+1

File: common/src/test/java/org/apache/falcon/entity/parser/FeedEntityParserTest.java (hunk 1 above)
Comment (ajayyadava) [inline common/src/test/java/org/apache/falcon/entity/parser/FeedEntityParserTest.java]: Hey @vramachan Thanks for fixing the docs and expected exceptions. Actually in this statement I meant something else. I am curious about the need for that Assert.fail statement. Even in failure scenarios, even without this statement we will get clear message that the ValidationException was expected but was not thrown. Anyhow, it is just nit picking :stuck_out_tongue:

**Comment by ajayyadava** [comment]
LGTM :+1:

**Pull request closed without merging**